pydash/numerical.py

```diff
@@ -383,6 +386,43 @@ def std_deviation(array):
     return math.sqrt(variance(array))
 
 
+def call_math_operator(value1, value2, callback):
+    """Return the result of the math operation on the given values.
+
+    Args:
+        value1 (int/float/str): Value passed by the user.
+        value2 (int/float/str): Value passed by the user.
+        callback (mixed, optional): Callback applied on the given values.
+
+    Returns:
+        float: Result after the math operation is conducted on the given
+            values.
+
+    Example:
+
+        >>> call_math_operator(5, 4, lambda x, y: x - y)
+        1.0
+        >>> call_math_operator(4, 6, lambda x, y: x * y)
+        24.0
+        >>> call_math_operator('5', '10', lambda x, y: x + y)
+        15.0
+        >>> call_math_operator('25', '5', lambda x, y: x / y)
+        5.0
+
+    .. versionadded:: TODO
+    """
+    # If the user has given any Nonetype values, we assign 0 as the default
+    # value to the values.
+    value1 = 0 if not value1 else value1
+    value2 = 0 if not value2 else value2
+
+    if isinstance(value1, str) or isinstance(value2, str):
+        value1 = float(value1)
+        value2 = float(value2)
+
+    return float(callback(value1, value2))
```

Comment:
No need to cast to `float` here. As long as both values are numbers, it's fine to let the operation return whatever the type would be (e.g. `int + int = int`, `int + float = float`, etc).